source/flow.hpp

```diff
@@ -6,28 +6,45 @@
 
 namespace sim {
 
-class Flow {
+class IFlow {
 public:
-    Flow(ISender *a_src, IReceiver *a_dest, float a_start_cwnd);
-
     // Start at time
-    void start(std::uint32_t time);
+    virtual void start(std::uint32_t time) = 0;
 
     // Try to generate a new packet if the internal state allows to do so.
-    // by placing it into the flow buffer of the source node.
-    // Schedule the next generation event.
-    bool try_to_generate(std::uint32_t packet_size);
+    // Returns true if a generate_event was created and placed in scheduler
+    virtual bool try_to_generate(std::uint32_t packet_size) = 0;
 
     // Update the internal state according to some congestion control algorithm
     // Call try_to_generate upon the update
-    void update();
+    virtual void update(std::uint32_t delay) = 0;
+
+    virtual ISender* get_sender() const noexcept = 0;
+    virtual IReceiver* get_receiver() const noexcept = 0;
+
+    virtual ~IFlow() = default;
+};
+
+class Flow : public IFlow {  // Fixed inheritance
+public:
+    Flow(ISender* a_src, IReceiver* a_dest, uint32_t a_packet_size);
+    void start(std::uint32_t time) override = 0;
+    bool try_to_generate(std::uint32_t packet_size) override = 0;
+    void update(std::uint32_t delay) override = 0;
```

Comment:
This functions are not virtual, so they cant be = 0
```suggestion
    void start(std::uint32_t time) override;
    bool try_to_generate(std::uint32_t packet_size) override;
    void update(std::uint32_t delay) override;
```